### PR TITLE
Used pyupgrade to remove compatibility code

### PR DESCRIPTION
**Pull request by ThatXliner**
Relates-to: #3860
# Pull Request Check List

Resolves: #issue-number-here ( may not remove all compatibility code)

- [N/A ] Added **tests** for changed code.
- [N/A ] Updated **documentation** for changed code.

### CODE REVIEW

**Comment by ThatXliner** [comment]
Sorry for the messy commits!

File: poetry/puzzle/solver.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def solve(self, use_latest: List[str] = None) -> List["OperationTypes"]:
                 )
                 self._provider.debug(
                     "Resolved with overrides: {}".format(
-                        ", ".join("({})".format(b) for b in self._overrides)
+                        ", ".join(f"({b})" for b in self._overrides)
```
Comment (finswimmer) [inline poetry/puzzle/solver.py:89]:
```py
f"Resolved with overrides: {', '.join(f'({b})' for b in self._overrides)}"
```

**Review by finswimmer — CHANGES_REQUESTED** [review]
Thanks a lot @ThatXliner :1st_place_medal: 

Some minor suggestion where the conversion to f-string haven't worked well enough.

TIL: `yield from` :+1:

File: poetry/publishing/uploader.py [hunk 2]
```diff
@@ -284,15 +280,15 @@ def _upload_file(
                     )
                 if dry_run or 200 <= resp.status_code < 300:
                     bar.set_format(
-                        " - Uploading <c1>{0}</c1> <fg=green>%percent%%</>".format(
+                        " - Uploading <c1>{}</c1> <fg=green>%percent%%</>".format(
```
[the remark this answers is not in the document]
Comment (ThatXliner) [reply poetry/publishing/uploader.py:283]: What's `%percent%%`

[the remark this answers is not in the document]
Comment (finswimmer) [reply poetry/publishing/uploader.py:283]: That's some magic `cleo` is doing: https://cleo.readthedocs.io/en/latest/helpers/progress_bar.html?highlight=percent#custom-formats

[the remark this answers is not in the document]
Comment (ThatXliner) [reply poetry/publishing/uploader.py:283]: So it's not [C-style string interpolation](https://docs.python.org/3/library/stdtypes.html#old-string-formatting), correct? In that case, I'll need to switch all the `"...{}...".format(..)` into f-strings

**Comment by ThatXliner** [comment]
For line 83
https://github.com/ThatXliner/poetry/blob/cdf6d9beafc84768de574cd5b906834ebfe8aa38/poetry/puzzle/solver.py#L83
Should we keep it like that?

**Review by finswimmer — APPROVED** [review]
Thanks a lot for your contribution. Looks good to me :+1:

**Pull request merged**